rye/src/cli/run.rs

```diff
@@ -29,7 +33,11 @@ enum Cmd {
 }
 
 pub fn execute(cmd: Args) -> Result<(), Error> {
-    let pyproject = PyProject::discover()?;
+    let pyproject = if let Some(pyproject_from) = cmd.pyproject_from {
+        PyProject::discover_from_path(pyproject_from)?
+    } else {
+        PyProject::discover()?
+    };
 
     // make sure we have the minimal virtualenv.
     sync(SyncOptions::python_only()).context("failed to sync ahead of run")?;
```

Comment:
Current revision doesn't always work correctly, because this sync call also uses PyProject::discover. The python_only machinery seems possible to work with, but updating all of sync to work with a "movable" pyproject.toml is not something I can do right now.